src/hooks/useFetch.js

```diff
@@ -0,0 +1,21 @@
+import axios from 'axios';
+
+const { REACT_APP_API_YOUTUBE_KEY } = process.env;
+
+const Fetch = async (search) => {
+  console.log(search);
+
+  const rep = await axios
+    .get(
+      `https://www.googleapis.com/youtube/v3/search?part=id&part=snippet&maxResults=5&key=${REACT_APP_API_YOUTUBE_KEY}`,
+      { ...search }
+    )
+    .then((response) => response.data)
+    .catch((error) => {
+      console.log(error);
+    });
```

Comment:
I recommend not combining async/await and promises in the same block, you should try to get the information using all async/await or using only promises.